### ISSUE TITLE
Can't extend Column type

### ISSUE THREAD

**Issue by RogerYK** [tobgu/qframe#32]
Hi, the dataframe library is wonerful. But I encounter some problems. The Column has some type int, string, float. But It is not enough. I find the Column interface, but it locate in internal package. I can't implement it to extend more type. 

What should I do if I want to save more type?

**Comment by tobgu** [comment]
Yeah, the Column type is really wide (a lot of methods), potentially in flux and still not ready to go public. I do have thoughts on publishing a smaller (adapter) interface that outsiders can implement though. Can you give the some input on the use case you have? It would be valuable to understand what your use case is to help in shaping such an API! Could you provide some details? Also, if you would like to have a go at this I'd be happy to guide you!

**Comment by RogerYK** [comment]
Ok, the database we use is Clickhouse. Its data type has json, array。for example.
id  |   name |  children |  actions
1   |    root   |   [2, 3]    |    {"time": "action"}
2   |    child  |   []          |    {"time": "action"}
3   |    child  |   []          |    {"time": "action"}
In addition, we also need save some data from a rpc. for example user data list.
id | user
1 |  User{name: "user_1"}
2 | User{name: "user_2"}
we need save these data in mem. and do calculate, sort, join.
I found that some columns only store data. It not need to implement those filter aplay function. 
What do you think?

**Comment by tobgu** [comment]
OK, so for your specific case do I understand it correctly that you would like to have a byte-container, more or less, for values to be included but not acted upon/processed? That would be a slightly different use case than the fully general column that I've also thought a bit about.
